apps/web/src/custom-pages/weekly-newsletter-page.tsx

```diff
@@ -18,23 +19,37 @@ import { LexicalSerializer } from "../components/lexical/lexical-serializer";
 import { TableOfContents } from "../components/table-of-contents";
 import { DateTime } from "../components/datetime";
 
-function NewsItemContent({ content }: { content?: EditorState }) {
-  if (!content) return null;
+async function NewsItemContent({ item }: { item: NewsItem }) {
+  const t = await getScopedI18n("weeklyNewsletter");
+  const content = item.content as unknown as EditorState;
 
   return (
     <div className="prose prose-headings:scroll-mt-40 prose-headings:xl:scroll-mt-24 max-w-prose hyphens-auto text-pretty">
+      {item.linkToSignUp ? (
+        <Link
+          // Do not create relative links
+          href={
+            item.linkToSignUp.startsWith("http")
+              ? item.linkToSignUp
+              : `//${item.linkToSignUp}`
+          }
+          target="_blank"
+          rel="noopener"
+        >
+          {t("link-to-sign-up")}{" "}
+          <span className="sr-only">{` ${t("for-event")} ${item.title}`}</span>
+        </Link>
+      ) : null}
       <LexicalSerializer nodes={content.root.children} />
```

Comment:
This might now fail if content is empty. Should probably wrap in a conditional :thinking: